From c55513e0f019384760a93ab5e8201e55b68abcad Mon Sep 17 00:00:00 2001
From: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Date: Thu, 21 Jan 2016 16:22:20 +0530
Subject: [PATCH 07/13] UBC/PD: Fix source cap certification issues.

This patch corrects the source capability certification issues.

Change-Id: I7fdf6cfd9982ce9c2fdf1cfee5af3e1bf2fdf7c6
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-15674
Signed-off-by: Venkataramana Kotakonda <venkataramana.kotakonda@intel.com>
Reviewed-on: https://android.intel.com:443/460777
---
 drivers/usb/typec/pd/policy_engine.c | 6 +++---
 drivers/usb/typec/pd/policy_engine.h | 2 +-
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/usb/typec/pd/policy_engine.c b/drivers/usb/typec/pd/policy_engine.c
index 2c66899..81562d8 100644
--- a/drivers/usb/typec/pd/policy_engine.c
+++ b/drivers/usb/typec/pd/policy_engine.c
@@ -1008,7 +1008,7 @@ static void pe_fetch_self_sink_cap(struct policy_engine *pe)
 	pdo[0].data_role_swap = FEATURE_SUPPORTED;
 	pdo[0].usb_comm = FEATURE_SUPPORTED;
 	pdo[0].ext_powered = FEATURE_NOT_SUPPORTED;
-	pdo[0].higher_cap = FEATURE_SUPPORTED;
+	pdo[0].higher_cap = FEATURE_NOT_SUPPORTED;
 	pdo[0].dual_role_pwr = FEATURE_SUPPORTED;
 
 	for (i = 0; i < pcaps.n_cap; i++) {
@@ -1846,9 +1846,9 @@ static void pe_process_state_pe_src_discovery(struct policy_engine *pe)
 		return;
 	}
 	if (pe->is_typec_port)
-		time_out = PE_TIME_TYPEC_SEND_SOURCE_CAP;
+		time_out = PE_TIME_TYPEC_SEND_SOURCE_CAP - PE_TIME_RECEIVE;
 	else
-		time_out = PE_TIME_SEND_SOURCE_CAP;
+		time_out = PE_TIME_SEND_SOURCE_CAP - PE_TIME_RECEIVE;
 	pe_start_timer(pe, SOURCE_CAPABILITY_TIMER, time_out);
 
 }
diff --git a/drivers/usb/typec/pd/policy_engine.h b/drivers/usb/typec/pd/policy_engine.h
index 58b4656..3268768 100644
--- a/drivers/usb/typec/pd/policy_engine.h
+++ b/drivers/usb/typec/pd/policy_engine.h
@@ -59,7 +59,7 @@
 #define PE_TIME_SOURCE_ACTIVITY		50
 #define PE_TIME_SWAP_SINK_READY		15
 #define PE_TIME_SWAP_SOURCE_START	20
-#define PE_TIME_TYPEC_SEND_SOURCE_CAP	200
+#define PE_TIME_TYPEC_SEND_SOURCE_CAP	180
 #define PE_TIME_TYPEC_SINK_WAIT_CAP	620
 #define PE_TIME_VCONN_SOURCE_OFF	25
 #define PE_TIME_VCONN_SOURCE_ON		100
-- 
1.9.1

